app/web/src/helpers/cadPackages/jsCad/jsCadController.ts

```diff
@@ -103,65 +135,31 @@ export const render: DefaultKernelExport['render'] = async ({
     const blob = new Blob([script], { type: 'text/javascript' })
     scriptWorker = new Worker(window.URL.createObjectURL(blob))
     let parameterDefinitions = []
-    scriptWorker.addEventListener('message', (e) => {
-      const data = e.data
+    scriptWorker.addEventListener('message', ({ data }) => {
       if (data.action == 'parameterDefinitions') {
         parameterDefinitions = data.data
       } else if (data.action == 'entities') {
         if (data.error) {
-          response = createUnhealthyResponse(new Date(), data.error)
+          workerHelper.resolver(createUnhealthyResponse(new Date(), data.error))
         } else {
-          response = createHealthyResponse({
-            type: 'geometry',
-            data: [...data.entities.map(CSG2Object3D).filter((o) => o)],
-            consoleMessage: data.scriptStats,
-            date: new Date(),
-            customizerParams: parameterDefinitions,
-            currentParameters,
-          })
+          workerHelper.resolver(
+            createHealthyResponse({
+              type: 'geometry',
+              data: data.entities.map(CSG2Object3D).filter((o) => o),
+              consoleMessage: data.scriptStats,
+              date: new Date(),
+              customizerParams: jsCadToCadhubParams(parameterDefinitions || []),
+            })
+          )
         }
-        callResolve()
       }
     })
 
-    callResolve()
-    response = null
+    workerHelper.resolver()
     scriptWorker.postMessage({ action: 'init', baseURI, alias: [] })
   }
 
-  if (
-    parameters &&
-    currentParameters &&
-    JSON.stringify(parameters) !== JSON.stringify(currentParameters)
-  ) {
-    // we are not evaluating code, but reacting to parameters change
-    scriptWorker.postMessage({
-      action: 'updateParams',
-      worker: 'script',
-      params: parameters,
-    })
-  } else {
-    scriptWorker.postMessage({
-      action: 'runScript',
-      worker: 'script',
-      script: code,
-      params: parameters || {},
-      url: 'jscad_script',
-    })
-  }
-  // we need this to keep the form filled with same data when new parameter definitions arrive
-  // each render of the script could provide new paramaters. In case some of them are still rpesent
-  // it is expected for them to stay the same and not just reset
-  currentParameters = parameters || {}
-
-  const waitResult = new Promise((resolve) => {
-    resolveReference = resolve
-  })
-
-  await waitResult
-  resolveReference = null
-  if (parameters) delete response.customizerParams
-  return response
```

Comment:
This is what I mean from this comment https://github.com/Irev-Dev/cadhub/pull/461/files#r692901725 about awaiting a promise just to return a different variable.

Reply:
yeah, that was messy. thanks